File: src/main/java/com/ject/studytrip/stamp/domain/repository/StampRepository.java
package com.ject.studytrip.stamp.domain.repository;

import com.ject.studytrip.stamp.domain.model.Stamp;
import java.util.List;
import java.util.Optional;

public interface StampRepository {
    Stamp save(Stamp stamp);

    List<Stamp> saveAll(List<Stamp> stamps);

    Optional<Stamp> findById(Long stampId);

    List<Stamp> findAllByIdIn(List<Long> ids);

    List<Stamp> findAllByTripIdAndDeletedAtIsNull(Long tripId);

    List<Stamp> findAllByTripIdOrderByCreatedAtAsc(Long tripId);
}

File: src/main/kotlin/com/ject/studytrip/stamp/domain/repository/StampRepository.kt
package com.ject.studytrip.stamp.domain.repository

import com.ject.studytrip.stamp.domain.model.Stamp
import java.util.Optional

interface StampRepository {
    fun save(stamp: Stamp): Stamp

    fun saveAll(stamps: List<Stamp>): List<Stamp>

    fun findById(stampId: Long): Optional<Stamp>

    fun findAllByIdIn(stampIds: List<Long>): List<Stamp>

    fun findAllByTripIdAndDeletedAtIsNull(tripId: Long): List<Stamp>

    fun findAllByTripIdOrderByCreatedAtAsc(tripId: Long): List<Stamp>
}
